Portfolio Mobile Version

Aries972 opened the pull request:
First try: It has been added the code for  rest of the screens of figma for template 2.
Third try: problem with white spaces ans responsiveness solved. Readme.md file improved. Red border removed.

KDlamini commented on index.html:
```diff
@@ -22,17 +22,146 @@
+                <button class="langbuttons2">html</button>
+                <button class="langbuttons2">bootstrap</button>
+                <button class="langbuttons2">Ruby</button>
+            </nav>
+            <button class="project2">See Project</button>
+        </div>
+
+        <div class="fourth-container">
+            <h3 class="profesional">Profesional Art Printing Data</h3>
+            <p class="par3">A daily selection of privately personalized reads; no accounts or sign-ups required. Has been the industry's standard</p>
+            <nav class="langs2">
+                <button class="langbuttons2">html</button>
+                <button class="langbuttons2">bootstrap</button>
+                <button class="langbuttons2">Ruby</button>
+            </nav>
+            <button class="project2">See Project</button>
+        </div>
+
+        <div class="fourth-container">
+            <h3 class="profesional">Profesional Art Printing Data</h3>
+            <p class="par3">A daily selection of privately personalized reads; no accounts or sign-ups required. Has been the industry's standard</p>
+            <nav class="langs2">
+                <button class="langbuttons2">html</button>
+                <button class="langbuttons2">bootstrap</button>
+                <button class="langbuttons2">Ruby</button>
+            </nav>
+            <button class="project2">See Project</button>
+        </div>
+
+        <div class="fourth-container">
+            <h3 class="profesional">Profesional Art Printing Data</h3>
+            <p class="par3">A daily selection of privately personalized reads; no accounts or sign-ups required. Has been the industry's standard</p>
+            <nav class="langs2">
+                <button class="langbuttons2">html</button>
+                <button class="langbuttons2">bootstrap</button>
+                <button class="langbuttons2">Ruby</button>
+            </nav>
+            <button class="project2">See Project</button>
+        </div>
+
```
Next, let's wrap all the project cards named `.fourth-container` into their own container. For illustration purposes, I'll name it `fourth-container-wrapper`:
![image](https://user-images.githubusercontent.com/38088818/164783498-4793c443-b0ec-44fb-ba30-3d3c18a6689e.png)

Now, let's use our `fourth-container-wrapper` to layout our `.fourth-container` child elements:
If I apply a `margin` of zero and a 100% `width`. The`red` and `green` borders are for illustration purposes.
![image](https://user-images.githubusercontent.com/38088818/164784150-481d8644-d46f-4a55-a163-0f27f01ef245.png)

The structure will look like this for now:
![image](https://user-images.githubusercontent.com/38088818/164784534-0a05d6ab-c28e-4450-b91e-e2b1f6d3db22.png)


Then I apply Flex properties and add `margin` zero on the `.fourth-container` as well:
![image](https://user-images.githubusercontent.com/38088818/164785220-acf6180d-8805-4d1f-b22f-291de1a294e4.png)

We now have this:
![image](https://user-images.githubusercontent.com/38088818/164785617-f782e634-8268-4d41-9b78-a532bf4789c3.png)

All these steps are of course my take on how I would go about resolving your display issues. You are more than welcome to do it in your own way and understanding. These are merely my suggestions, however, the final product should be the same: which is making sure that your app looks as close to the Figma design as possible.

Good luck and happy coding 🥂

Aries972 replied:
About the fourth-container-wrapper, I did not implement it because when I checked the page it was already centered. I think this is because the corrections made in step 2. If you feel I still should do it please let me know it won't be a problem.

KDlamini replied:
That's alright, I did mention that some of the listed suggestions are merely my take on structuring your elements. With the exception of grouping your sections in `section` tag and using a `ul` tag for listing technologies; which are part of the project requirements. You can do it in your own way and understanding 💯

KDlamini commented on index.html:
```diff
@@ -22,17 +22,146 @@
+            <img src="images/placeholder.png" alt="placeholder" class="img2">
+            <h3 id="multi">Multi-Post Sories</h3>
+            <p id="par2">A daily selection of privately personalized reads; no accounts or sign-ups required. Has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a standard dummy text.</p>
+            <nav class="langs">
+                <button class="langbuttons">css</button>
+                <button class="langbuttons">html</button>
+                <button class="langbuttons">bootstrap</button>
+                <button class="langbuttons">Ruby</button>
+            </nav>
```
Kindly use a list `ul` and each of the project technologies (e.g., "JavasScript", "HTML", and "CSS")  should be a `li` element because it is semantic that way. The `nav` element should only be used for the `navbar` menu items since they are used for `navigation`, hence `nav`.

Aries972 replied:
I implemented ul and li where requested, but I have a little problem. The elements tend to go more to the right. I think it is because of the bullet point. Please if you can help me to correct this I will be grateful.

KDlamini replied:
You can use a `padding: 0;` on the parent `ul` tag to remove the default indentation of `li` items.

KDlamini commented on index.html:
```diff
@@ -22,17 +22,146 @@
+            <button class="project">See Project</button>
+        </div>
+
+        <div class="fourth-container">
+            <h3 class="profesional">Profesional Art Printing Data</h3>
+            <p class="par3">A daily selection of privately personalized reads; no accounts or sign-ups required. Has been the industry's standard</p>
+            <nav class="langs2">
+                <button class="langbuttons2">html</button>
+                <button class="langbuttons2">bootstrap</button>
+                <button class="langbuttons2">Ruby</button>
+            </nav>
+            <button class="project2">See Project</button>
+        </div>
+
+        <div class="fourth-container">
+            <h3 class="profesional">Profesional Art Printing Data</h3>
+            <p class="par3">A daily selection of privately personalized reads; no accounts or sign-ups required. Has been the industry's standard</p>
+            <nav class="langs2">
+                <button class="langbuttons2">html</button>
+                <button class="langbuttons2">bootstrap</button>
+                <button class="langbuttons2">Ruby</button>
+            </nav>
+            <button class="project2">See Project</button>
+        </div>
+
+        <div class="fourth-container">
+            <h3 class="profesional">Profesional Art Printing Data</h3>
+            <p class="par3">A daily selection of privately personalized reads; no accounts or sign-ups required. Has been the industry's standard</p>
+            <nav class="langs2">
+                <button class="langbuttons2">html</button>
+                <button class="langbuttons2">bootstrap</button>
+                <button class="langbuttons2">Ruby</button>
+            </nav>
+            <button class="project2">See Project</button>
+        </div>
```
Same this applies to all your `fourth-container` elements:

Kindly use a list `ul` and each of the project technologies (e.g., "JavasScript", "HTML", and "CSS")  should be a `li` element because it is semantic that way. The `nav` element should only be used for the `navbar` menu items since they are used for `navigation`, hence `nav`.

Aries972 replied:
I implemented ul and li where requested, but I have a little problem. The elements tend to go more to the right. I think it is because of the bullet point. Please if you can help me to correct this I will be grateful.

KDlamini replied:
same here, you can use a `padding: 0;` on the parent `ul` tag to remove the default indentation of `li` items.

KDlamini reviewed (changes requested):
### Hi @Aries972 🙋‍♀️,  

Excellent work so far! 
You have made a great effort creating the works and about sections according to the Figma design. 🥳 
There's always room for improvement. There are some issues that you still need to work on improving your project according to the requirements. You are almost there! 💪🏾 

## Highlights 🌞 
✔️  Correct PR title
✔️  Correct use of Gitflow 
✔️  No linter errors

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Aries972 commented on index.html:
```diff
@@ -22,17 +22,146 @@
+                <li><img src="images/eme.png" alt="eme" class="img1"></li>
             </ul>
         </div>
+
+        <div class="second-container">
+            <h2 id="recent">My Recent Works</h2>
+        </div>
+
```
I did almost all the suggestions, I did not remove width in mobile-header and .content (I renamed it to first-container), but I think the blank lines were eliminated. Please check it.

Aries972 commented on index.html:
```diff
@@ -22,17 +22,146 @@
+                <button class="langbuttons2">html</button>
+                <button class="langbuttons2">bootstrap</button>
+                <button class="langbuttons2">Ruby</button>
+            </nav>
+            <button class="project2">See Project</button>
+        </div>
+
+        <div class="fourth-container">
+            <h3 class="profesional">Profesional Art Printing Data</h3>
+            <p class="par3">A daily selection of privately personalized reads; no accounts or sign-ups required. Has been the industry's standard</p>
+            <nav class="langs2">
+                <button class="langbuttons2">html</button>
+                <button class="langbuttons2">bootstrap</button>
+                <button class="langbuttons2">Ruby</button>
+            </nav>
+            <button class="project2">See Project</button>
+        </div>
+
+        <div class="fourth-container">
+            <h3 class="profesional">Profesional Art Printing Data</h3>
+            <p class="par3">A daily selection of privately personalized reads; no accounts or sign-ups required. Has been the industry's standard</p>
+            <nav class="langs2">
+                <button class="langbuttons2">html</button>
+                <button class="langbuttons2">bootstrap</button>
+                <button class="langbuttons2">Ruby</button>
+            </nav>
+            <button class="project2">See Project</button>
+        </div>
+
+        <div class="fourth-container">
+            <h3 class="profesional">Profesional Art Printing Data</h3>
+            <p class="par3">A daily selection of privately personalized reads; no accounts or sign-ups required. Has been the industry's standard</p>
+            <nav class="langs2">
+                <button class="langbuttons2">html</button>
+                <button class="langbuttons2">bootstrap</button>
+                <button class="langbuttons2">Ruby</button>
+            </nav>
+            <button class="project2">See Project</button>
+        </div>
+
```
I included all the section tag. If something is not good or missing please let me know. I also followed your instructions to implement a new section class called works and been able to remove the width property. Thanks a lot!!!

KDlamini replied:
Great job! 👍🏾

Aries972 commented:
First of all, thank you very much for your kind revision.
1. I did almost all the suggestions, I did not remove width in mobile-header and .content (I renamed it to first-container), but I think the blank lines were eliminated. Please check it.
2. I included all the section tag. If something is not good or missing please let me know. I also followed your instructions to implement a new section class called works and been able to remove the width property. 
3. About the fourth-container-wrapper, I did not implement because when I checked the page it was already centered. I think this is because the corrections made in step 2. If you feel I still should do it please let me know it won't be a problem.
4-5. I implemented ul and li where requested, but I have a little problem. The elements tend to go more to the right. I think it is because of the bullet point. Please if you can help me to correct this I will be grateful.

*I maintained a red border in order to verify the alignment. I will remove it later.

mahbubul14 commented on styles.css:
```diff
@@ -1,16 +1,23 @@
 
 .main-container {
+  display: grid;
+  grid-template-columns: auto;
+  grid-template-rows: 10;
+  align-items: center;
   width: 375px;
-  display: flex;
-  flex-direction: column;
+  margin: 0;
+  border: 2px solid red;
 }
```
- [ ] There is still white space on the right and left sides of your page. The page is not also optimized for the mobile version well. You can use [CSS Media Queries](https://www.w3schools.com/css/css3_mediaqueries.asp) for your mobile version to make it responsive. :+1: 

- [ ] Kindly remove the red border from your page. :+1: 

https://user-images.githubusercontent.com/70483835/164817852-25bc3e91-7e0f-407f-9235-177b44b86955.mp4

Aries972 replied:
Red border removed.
Problem with whitespaces and responsiveness I think is solved, but more corrections are needed please let me know.
Readme.md file was improved.
Thank you very much for your kind feedback. I am learning a lot.

mahbubul14 reviewed (changes requested):
Hi @Aries972 ,

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

# Required Changes ♻️

- [x]  Currently, your `README` file is not updated well enough. Please update your `README` file. You can use [this template ](https://github.com/microverseinc/readme-template)for the `README` but remember to customize it to your project.

- [ ] Please check the inline comments under the review for more suggestions. :arrow_down: 

## **Note: You can tag the reviewer if you have any questions. If you don't tag the reviewer, he won't get notified and not be able to reply back.**

###  To Highlights
:heavy_check_mark: No linters error.
:heavy_check_mark: Github flow used properly.
:heavy_check_mark: Descriptive PR title & description.

### Optional suggestions

- [ ] The commit message should be written in the imperative tense and the first letter should be capitalized. `Ex: "Fix bug" instead of "Bug correction".` [For more details.](https://www.freecodecamp.org/news/level-you-up-to-awesome-commit-messages-a85558cb90e8)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also reach me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

KDlamini commented:
> First of all, thank you very much for your kind revision.
> 
> 1. I did almost all the suggestions, I did not remove width in mobile-header and .content (I renamed it to first-container), but I think the blank lines were eliminated. Please check it.
> 2. I included all the section tag. If something is not good or missing please let me know. I also followed your instructions to implement a new section class called works and been able to remove the width property.
> 3. About the fourth-container-wrapper, I did not implement because when I checked the page it was already centered. I think this is because the corrections made in step 2. If you feel I still should do it please let me know it won't be a problem.
>    4-5. I implemented ul and li where requested, but I have a little problem. The elements tend to go more to the right. I think it is because of the bullet point. Please if you can help me to correct this I will be grateful.
> 
> *I maintained a red border in order to verify the alignment. I will remove it later.

Hi @Aries972

I apologize for the late reply, it was quite very late in my time zone when you made your comment.
Great job implementing most of my suggestions. I made individual replies to each comment to respond to your questions.

Cheers and happy coding 🥂

AshakaE commented:
Hi @Aries972 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Aries972 commented:
Thank you very much for all your help @KDlamini, @AshakaE, @mahbubul14.
